File: 단속카메라.java
import java.util.*;
class Solution {
    public int solution(int[][] routes) {
        int answer = 1;
        Arrays.sort(routes, (num1, num2)-> Integer.compare(num1[1], num2[1])); // 2차원배열 두번째 키값을 기준으로 오름차순 정렬
  
        int camera = routes[0][1];
        for (int i=1; i<routes.length;i++){
            if (camera<routes[i][0]){
                camera = routes[i][1];
                answer++;
            }
        }
        return answer;
    }
}

File: 단속카메라.py
from collections import deque
def solution(routes):
    answer = 1
    routes.sort(key = lambda x:x[-1]) #정렬 이렇게 꼭 해줘야 함...
    routes = deque(routes)
    q = deque()
    q.append(routes.popleft())
    while routes:
        top_end = q[-1][1]
        push_start, push_end = routes.popleft()
        if push_start> top_end:
            q.pop()
            answer+=1
            q.append([push_start, push_end])
    return answer
